### PR TITLE
feat: add generic marshalJSON function

### PR DESCRIPTION
**Pull request by yashmehrotra**

### CODE REVIEW

File: celext/cel_test.go [hunk 1]
```diff
@@ -119,10 +119,10 @@ func TestCelJSON(t *testing.T) {
 		Input  string
 		Output any
 	}{
-		{Input: `dyn([{'name': 'John', 'age': 30}]).toJSONArray()`, Output: `[{"age":30,"name":"John"}]`},
-		{Input: `[{'name': 'John'}].toJSONArray()`, Output: `[{"name":"John"}]`},
-		{Input: `dyn({'name': 'John'}).toJSON()`, Output: `{"name":"John"}`},
-		{Input: `{'name': 'John'}.toJSON()`, Output: `{"name":"John"}`},
+		{Input: `dyn([{'name': 'John', 'age': 30}]).marshalJSON()`, Output: `[{"age":30,"name":"John"}]`},
```
Comment (moshloop) [inline celext/cel_test.go:122]: Why not just toJSON ?

Comment (yashmehrotra) [reply celext/cel_test.go:122]: just to replicate the go terminology, should I change it to `toJSON` ?

Comment (moshloop) [reply celext/cel_test.go:122]: yeh, because we have a few other functions toYAML|TOML|CSV etc..

Comment (yashmehrotra) [reply celext/cel_test.go:122]: cool, will update

**Pull request merged**